test/unit/github_fetcher/issue_comments_test.rb

```diff
@@ -0,0 +1,55 @@
+require 'test_helper'
+
+class GithubFetcher::IssueCommentsTest < ActiveSupport::TestCase
+  def comments_fetcher
+    issue = issues(:issue_three)
+    issue.repo = repos(:sinatra_sinatra) # For some reason, this isn't set :/
+
+    GithubFetcher::IssueComments.new(
+      owner_name: issue.owner_name,
+      repo_name: issue.repo_name,
+      number: issue.number
+    )
+  end
+
+  test "quacks like a GithubFetcher::Resource" do
+    fetcher = comments_fetcher
+    GithubFetcher::Resource.instance_methods(false).each do |method|
+      assert fetcher.respond_to? method, "Failed to respond_to? #{method}"
+    end
+    GithubFetcher::Resource.private_instance_methods(false).each do |method|
+      assert fetcher.respond_to? method, "Failed to respond_to? #{method}"
+    end
+  end
+
+  test "#as_json doesn't raise errors" do
```

Comment:
Yeah I go back and forth on that. In this case, part of the purpose of the extraction is to not raise this particular error at least, so it felt worthwhile. But I can remove these if you don't think they add enough value